x/auction/keeper/querier.go

```diff
@@ -1,151 +1,147 @@
+			return queryAuctions(ctx, req, keeper, legacyQuerierCdc)
 		case types.QueryGetParams:
-			return queryGetParams(ctx, req, keeper)
+			return queryGetParams(ctx, req, keeper, legacyQuerierCdc)
 		case types.QueryNextAuctionID:
-			return queryNextAuctionID(ctx, req, keeper)
+			return queryNextAuctionID(ctx, req, keeper, legacyQuerierCdc)
 		default:
 			return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unknown %s query endpoint", types.ModuleName)
 		}
 	}
 }
 
-func queryAuction(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) ([]byte, error) {
-	// Decode request
-	var requestParams types.QueryAuctionParams
-	err := types.ModuleCdc.UnmarshalJSON(req.Data, &requestParams)
+// query params in the auction store
+func queryGetParams(ctx sdk.Context, req abci.RequestQuery, keeper Keeper, legacyQuerierCdc *codec.LegacyAmino) ([]byte, error) {
+	// Get params
+	params := keeper.GetParams(ctx)
+
+	// Encode results
+	bz, err := codec.MarshalJSONIndent(legacyQuerierCdc, params)
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrJSONMarshal, err.Error())
+	}
+
+	return bz, nil
+}
+
+func queryAuction(ctx sdk.Context, req abci.RequestQuery, k Keeper, legacyQuerierCdc *codec.LegacyAmino) ([]byte, error) {
+	var params types.QueryAuctionParams
+	err := legacyQuerierCdc.UnmarshalJSON(req.Data, &params)
 	if err != nil {
 		return nil, sdkerrors.Wrap(sdkerrors.ErrJSONUnmarshal, err.Error())
 	}
 
-	// Lookup auction
-	auction, found := keeper.GetAuction(ctx, requestParams.AuctionID)
-	if !found {
-		return nil, sdkerrors.Wrapf(types.ErrAuctionNotFound, "%d", requestParams.AuctionID)
+	auction, ok := k.GetAuction(ctx, params.AuctionID)
+	if !ok {
+		return nil, sdkerrors.Wrap(types.ErrAuctionNotFound, fmt.Sprintf("%d", params.AuctionID))
 	}
 
-	// Encode results
-	bz, err := codec.MarshalJSONIndent(keeper.cdc, auction)
+	bz, err := codec.MarshalJSONIndent(legacyQuerierCdc, auction)
 	if err != nil {
 		return nil, sdkerrors.Wrap(sdkerrors.ErrJSONMarshal, err.Error())
 	}
-
 	return bz, nil
 }
 
-func queryAuctions(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) ([]byte, error) {
+func queryAuctions(ctx sdk.Context, req abci.RequestQuery, k Keeper, legacyQuerierCdc *codec.LegacyAmino) ([]byte, error) {
 	var params types.QueryAllAuctionParams
-	err := types.ModuleCdc.UnmarshalJSON(req.Data, &params)
+	err := legacyQuerierCdc.UnmarshalJSON(req.Data, &params)
 	if err != nil {
 		return nil, sdkerrors.Wrap(sdkerrors.ErrJSONUnmarshal, err.Error())
 	}
 
-	unfilteredAuctions := keeper.GetAllAuctions(ctx)
-	auctions := filterAuctions(ctx, unfilteredAuctions, params)
+	unfilteredAuctions := k.GetAllAuctions(ctx)
+	auctions := filterAuctions(ctx, unfilteredAuctions, params, legacyQuerierCdc)
 	if auctions == nil {
-		auctions = types.Auctions{}
+		auctions = []types.Auction{}
 	}
 
-	bz, err := codec.MarshalJSONIndent(keeper.cdc, auctions)
+	res, err := codec.MarshalJSONIndent(legacyQuerierCdc, unfilteredAuctions)
```

Comment:
Should this return `auctions` rather than `unfilteredAuctions`? Looks like it's bypassing the filter.

Reply:
Updated in https://github.com/Kava-Labs/kava/pull/1058/commits/a61fb18b4a29fcbc6a125c78b4fdbc9af1076047